src/NuGet.Core/NuGet.Common/PublicAPI.Unshipped.txt

```diff
@@ -7,4 +7,7 @@ NuGet.Common.NuGetLogCode.NU1903 = 1903 -> NuGet.Common.NuGetLogCode
 NuGet.Common.NuGetLogCode.NU1904 = 1904 -> NuGet.Common.NuGetLogCode
 NuGet.Common.NuGetLogCode.NU1905 = 1905 -> NuGet.Common.NuGetLogCode
 NuGet.Common.NuGetLogCode.NU3042 = 3042 -> NuGet.Common.NuGetLogCode
+NuGet.Common.NuGetLogCodeUtility
 NuGet.Common.PathResolver.SearchPathResult.SearchPathResult() -> void
+static NuGet.Common.NuGetLogCodeUtility.IsVulnerabilityCode(NuGet.Common.NuGetLogCode nuGetLogCode) -> bool
+~static NuGet.Common.NuGetLogCodeUtility.AreVulnerabilitiesInCodes(System.Collections.Generic.IEnumerable<NuGet.Common.NuGetLogCode> nugetLogCodes) -> bool
```

Comment:
There was a thread suggesting we not introduce a public API. What about creating an internal API under `NuGet.SolutionRestoreManager` as an alternative? Basically, just moving these new APIs there, then this PR won't need to modify NuGet.Common at all.

Reply:
My guess is that Martin was asked to create a public API that others teams could use. The problem is that we don't make `NuGet.Common.dll` available (don't provide a binding redirect) in VS, so any team who would want to use this public API, would need to start shipping NuGet.Common.dll in their own vsix, and they'll be out of date each time we insert a newer version of NuGet into VS. So, creating this public API for other VS components is well intentioned, but not entirely useful. However, apps that are not VS would be able to use this API just fine. It's just a question of whether the API name is good enough.

The .NET project system team emailed me a few weeks (months?) ago about all the package vulnerability feature, and I replied with questions asking if they're going to be making code changes only in the solution explorer code that exists in the NuGet.Client repo, or if they need to call APIs from other repos, and also what information they have (assets file? msbuild logger codes? only package id?) so we can try to design an API that works. They never got back to me, so I don't have enough information to design a good API for them.

So, I agree that unless we currently have someone asking for a public API, and we know what constraints they have (can they actually use NuGet.Common.dll?), that maybe not adding a new public API now is safer. But a method that takes a `NuGetLogCode` and returns a `bool` if the code is one of the package vulnerability, is an API that I can conceptually agree to. I'd like to improve the class & method names though, they don't quite "feel" right to me at the moment. I don't like the API that takes an `IEnumerable<NuGetLogCode>` though.

Reply:
As far as naming, then it sounds like Andy agrees with my suggestion above: move the API, make it internal. As for naming, I don't see a suggestion other than to rename it. I'd think something like
`HasVunerabilityLogCode` or `ContainsVulnerableCode`?..

Reply:
Nevermind, he did suggest some names which look good to me in another thread: https://github.com/NuGet/NuGet.Client/pull/5258#discussion_r1273166694